internal/intentapi/apiserver_proxy.go

```diff
@@ -0,0 +1,71 @@
+package intentapi
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/grafana/grafana/pkg/infra/log"
+	"github.com/grafana/grafana/pkg/setting"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/kubectl/pkg/proxy"
+)
+
+// ApiserverProxyConfig is the config for ApiserverProxy.
+type ApiserverProxyConfig struct {
+	RestCfg          *rest.Config
+	KeepaliveTimeout time.Duration
+}
+
+// NewApiserverProxyConfig parses and returns a new ApiserverProxyConfig.
+func NewApiserverProxyConfig(cfg *setting.Cfg) (ApiserverProxyConfig, error) {
+	sec := cfg.Raw.Section("intentapi.proxy")
+
+	configPath := sec.Key("config_path").MustString("")
+	targetUrl := sec.Key("target_url").MustString("https://127.0.0.1:6443")
+
+	// use the current context in kubeconfig
+	config, err := clientcmd.BuildConfigFromFlags("", configPath)
+	if err != nil {
+		return ApiserverProxyConfig{}, err
+	}
+
+	config.Host = targetUrl
+
+	return ApiserverProxyConfig{
+		RestCfg:          config,
+		KeepaliveTimeout: sec.Key("keepalive_timeout").MustDuration(1 * time.Minute),
+	}, nil
+}
+
+// ApiserverProxy is a proxy for kube-apiserver.
+type ApiserverProxy struct {
+	config  ApiserverProxyConfig
+	handler http.Handler
+	logger  log.Logger
+}
+
+// ProvideApiserverProxy provides a new ApiserverProxy with given configuration.
+func ProvideApiserverProxy(cfg *setting.Cfg) (*ApiserverProxy, error) {
+	conf, err := NewApiserverProxyConfig(cfg)
+	if err != nil {
+		return nil, err
+	}
+
+	proxy, err := proxy.NewProxyHandler("/", nil, conf.RestCfg, conf.KeepaliveTimeout, false)
```

Comment:
Hm, I think this should be possible, but I don't see why we would need that, since we are already running the server on a separate port.

@sdboyer what do you think?

Reply:
yeah, separate port is sufficient. my subpath idea was early stages, and is fully superseded by doing things on a separate port